### PR TITLE
Allow users to configure invenio communities

### LINKED ISSUE #157: Configure Invenio communities for publication
Invenio-based platforms allow a kind of curation via "Communities" in which publications can be listed. Currently, hermes does not have the ability to add communities to a publication. This should be made possible via a configuration option for the deposition plugin.

### PR DESCRIPTION
**Pull request by zyzzyxdonta**
Closes #157

### CODE REVIEW

File: src/hermes/commands/deposit/invenio.py [hunk 1]
```diff
@@ -68,6 +69,13 @@ def prepare_deposit(click_ctx: click.Context, ctx: CodeMetaContext):
     license = _get_license_identifier(ctx, licenses_api_url)
     ctx.update(invenio_path["license"], license)
 
+    communities_api_path = invenio_config.get("api_paths", {}).get(
+        "communities", _DEFAULT_COMMUNITIES_API_PATH
+    )
+    communities_api_url = f"{site_url}/{communities_api_path}"
+    communities = _get_community_identifiers(ctx, communities_api_url)
+    ctx.update(invenio_path["communities"], communities)
```
Comment (led02) [inline src/hermes/commands/deposit/invenio.py:77]: **nitpick:** I smell some duplicate code here... Probably fine for now, might be subject to further abstraction later on.

File: src/hermes/commands/deposit/invenio.py [hunk 2]
```diff
@@ -438,3 +446,34 @@ def _get_license_identifier(ctx: CodeMetaContext, license_api_url: str):
...
+    identifiers can not be found on the site, a :class:`MisconfigurationError` is
+    raised.
+    """
+
+    communities = config.get("deposit").get("invenio", {}).get("communities")
+    if communities is None:
+        return None
+
+    session = requests.Session()
+    session.headers = {"User-Agent": hermes_user_agent}
```
Comment (led02) [inline src/hermes/commands/deposit/invenio.py:465]: **question:** I like the idea of a custom user agent... maybe we should move the network handling code to a common submodule (however, out-of-scope for poc).

**Review by led02 — APPROVED** [review]
Some potential improvement for later versions exists, but this is fine for now.

However, we should consider some additional refactorings for *beta*:
- Network helpers API
- Better configuration file handling (e.g., providing defaults etc.)
- Common API for checking / matching against a ReST endpoint

File: src/hermes/commands/deposit/invenio.py (hunk 1 above)
Comment (zyzzyxdonta) [reply src/hermes/commands/deposit/invenio.py:77]: Yes, I noticed this, too. Even in the `_get_community/license_identifier(s)` there are some duplications

File: src/hermes/commands/deposit/invenio.py (hunk 2 above)
Comment (zyzzyxdonta) [reply src/hermes/commands/deposit/invenio.py:465]: Agreed. This is the second time this has come up so I'll write an issue for this.

Comment (zyzzyxdonta) [reply src/hermes/commands/deposit/invenio.py:465]: https://github.com/hermes-hmc/workflow/issues/165

**Pull request merged**